src/fallbacks.jl

```diff
@@ -4,12 +4,16 @@
 
 # Turn any AbstractColumns into an AbstractRow iterator
 
-# get the number of rows in the incoming table
-function rowcount(cols)
+"Return the number of rows in the incoming table."
+function nrow(cols)
     names = columnnames(cols)
     isempty(names) && return 0
     return length(getcolumn(cols, names[1]))
 end
+@deprecate rowcount(x) nrow(x)
+
+"Return the number of columns in the incoming table."
+ncol(cols) = length(columnnames(cols))
```

Comment:
Same here; `columnnames` and `getcolumn` are only technically allowed to be called on the result of `Tables.columns` or on individual iterated rows from `Tables.rows`, so this makes me a bit uncomfortable.